vmdb/app/models/miq_expression.rb

```diff
@@ -1209,7 +1209,8 @@ def self.quote(val, typ, mode=:ruby)
     case typ.to_s
     when "string", "text", "boolean", nil
       val = "" if val.nil? # treat nil value as empty string
-      return mode == :sql ? ActiveRecord::Base.connection.quote(val) : "'" + val.to_s.gsub(/'/, "\\\\'") + "'" # escape any embedded single quotes
+      # escape any embedded single quotes, etc. - needs to be able to handle even values with trailing backslash
+      return mode == :sql ? ActiveRecord::Base.connection.quote(val) : val.to_s.inspect
```

Comment:
As far as I can tell, the resulting string is always only passed to eval, so assuming `inspect` quotes everything properly, it shouldn't matter. But yes, I do intend to add tests for it, that's what the WIP is for :)